src/chi-vue/src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,145 @@
+    );
+  }
+
+  show() {
+    this.$emit(TOOLTIP_EVENTS.SHOW);
+    if (this._animation && !this._animation.isStopped()) {
+      this._animation.stop();
+    }
+
+    if (this._tooltipElementNode && !this._shown) {
+        this._shown = true;
+        this._animation = ThreeStepsAnimation.animationFactory(
+          () => {
+            this._tooltipElementNode.classList.add(TRANSITIONING_CLASS);
+          },
+          () => {
+            this._tooltipElementNode.classList.add(ACTIVE_CLASS);
+          },
+          () => {
+            this._tooltipElementNode.classList.remove(TRANSITIONING_CLASS);
+            this.$emit(TOOLTIP_EVENTS.SHOWN);
+          },
+          ANIMATION_DURATION.SHORT
+        );
+      }
+  }
+
+  hide() {
+    this.$emit(TOOLTIP_EVENTS.HIDE);
+    if (this._animationTimeout) {
+      clearTimeout(this._animationTimeout);
+    }
+    if (this._animation && !this._animation.isStopped()) {
+      this._animation.stop();
+    }
+
+    if (this._tooltipElementNode && this._shown) {
+        this._shown = false;
+        this._animation = ThreeStepsAnimation.animationFactory(
+          () => {
+            this._tooltipElementNode.classList.add(TRANSITIONING_CLASS);
+          },
+          () => {
+            this._tooltipElementNode.classList.remove(ACTIVE_CLASS);
+          },
+          () => {
+            this._tooltipElementNode.classList.remove(TRANSITIONING_CLASS);
+            this.$emit(TOOLTIP_EVENTS.HIDDEN);
+          },
+          ANIMATION_DURATION.SHORT
+        );
+    }
+  }
+
+  mounted() {
+    const slotTooltipTriggerElements = this.$slots.default;
+    const tooltipElementNode = document.getElementById(this._uuid);
+
+    if (tooltipElementNode) {
+      this._tooltipElementNode = tooltipElementNode;
+    }
+
+    if (slotTooltipTriggerElements) {
+      slotTooltipTriggerElements.forEach(vnode => {
+        const slotElement = vnode.elm;
+        const triggerShow = () => {
+          this._animationTimeout = window.setTimeout(() => {
+            if (!this._shown) {
+              this.show();
+            }
+          }, ANIMATION_DELAY);
+        };
+        const triggerHide = () => {
+          this.hide();
+        }
+
+        createPopper(slotElement as Element, this._tooltipElementNode, {
+          placement: this.position,
+          modifiers: [
+            {
+              name: 'offset',
+              options: {
+                offset: [0, 8]
+              }
+            },
+          ]
+        });
+
+        if (slotElement) {
+          slotElement.addEventListener('mouseenter', triggerShow.bind(this));
+          slotElement.addEventListener('mouseleave', triggerHide.bind(this));
+          slotElement.addEventListener('focus', triggerShow.bind(this));
+          slotElement.addEventListener('blur', triggerHide.bind(this));
+        }
+      });
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        {this.$slots.default}
+        {this._tooltipElement}
```

Comment:
My recommendation is that you write the tooltipElement full code here in the template (of in a factory method that returns the template for the tooltip that is re-rendered each time the method is called). Any changes in properties will lead into a re-render and will be updated.